example_app/src/app.rs

```diff
@@ -8,37 +11,44 @@ pub struct ExampleApp {
     frontend: Option<FrontEnd>,
 }
 
-impl epi::App for ExampleApp {
-    fn name(&self) -> &str {
-        "ewebsocket example app"
+impl ExampleApp {
+    pub fn new(cc: &eframe::CreationContext<'_>) -> Self {
+        let mut example_app = ExampleApp {
+            url: "wss://echo.websocket.events/.ws".into(),
+            ..Default::default()
+        };
+
+        example_app.connect(cc.egui_ctx.clone());
+
+        Self::default()
     }
 
-    fn setup(
-        &mut self,
-        _ctx: &egui::Context,
-        frame: &epi::Frame,
-        _storage: Option<&dyn epi::Storage>,
-    ) {
-        if let Some(web_info) = &frame.info().web_info {
-            // allow `?url=` query param
-            if let Some(url) = web_info.location.query_map.get("url") {
-                self.url = url.clone();
+    fn connect(&mut self, ctx: Context) {
```

Comment:
I've let this function take a moved `Context`, and everywhere that calls this function clones `Context` first. Is that the right way to handle passing context? If I just pass a reference, then I run into an issue with lifetimes that I didn't want to explore.

Is `Context` designed to be cloned?